tests/test_partial.py

```diff
@@ -82,6 +84,22 @@ class VanillaDataClassPersonWithClassVar(VanillaDataClassPerson):
             assert NoneType not in get_args(annotation)
 
 
+def test_partial_msgspec_struct() -> None:
+    class MsgspecPersonWithClassVar(MsgSpecStructPerson):
+        cls_var: ClassVar[int]
+
+    partial = Partial[MsgspecPersonWithClassVar]
+
+    assert len(type_info(MsgspecPersonWithClassVar).fields) == len(type_info(MsgSpecStructPerson).fields)  # type: ignore
```

Comment:
You might want to use `msgspec.structs.fields` ([docs](https://jcristharif.com/msgspec/api.html#msgspec.structs.fields)) here instead of `type_info`. The whole `msgspec.inspect` module is still marked as experimental, while `msgspec.structs.fields` is solid and not going anywhere. It also gets you the same info that you want here.

Reply:
Cool, thanks